### PR TITLE
Cleanup timers so process shuts down cleanly

### PR DESCRIPTION
**Pull request by arlolra**
Part of T158248

/cc @wikimedia-services

### CODE REVIEW

**Comment by d00rman** [comment]
Could also use a version bump so that it can be picked up in service-runner and upper deps.

**Comment by arlolra** [comment]
Will do when I get this out of WIP

**Comment by arlolra** [comment]
Needed for https://github.com/wikimedia/service-runner/pull/150

**Comment by Pchelolo** [comment]
LGTM, @gwicke as an author of the library could you have a look to? it's now pretty pressing since it's needed for another PR.

File: index.js [hunk 1]
```diff
@@ -90,14 +93,20 @@ Limitation.prototype.setup = function() {
...
-        setTimeout(function() {
+        self._globalUpdatesTimeout = setTimeout(function() {
             return self._globalUpdates();
         }, self._getRandomizedInterval(0.5));
 
         return self;
     });
 };
 
+Limitation.prototype.cleanup = function() {
```
Comment (gwicke) [inline index.js]: Minor: Perhaps name this `close`, in line with various servers, sockets etc?

**Comment by gwicke** [comment]
LGTM overall. Minor naming proposal inline.

We should also document this method in the README.

File: index.js (hunk 1 above)
Comment (gwicke) [reply index.js]: Or `stop()`, in line with `ServiceRunner.stop()`

Comment (arlolra) [reply index.js]: Ok, I went with `stop()`.

**Comment by arlolra** [comment]
> We should also document this method in the README.

Added a line for that.

**Pull request merged**